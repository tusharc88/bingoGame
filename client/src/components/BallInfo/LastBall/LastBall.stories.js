import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered';
import { withKnobs, number, text } from '@storybook/addon-knobs';
import LastBall from './index';

storiesOf('BallInfo', module)
  .addDecorator(withKnobs)
  .addDecorator(centered)
  .addWithInfo(
    'LastBall',
    <LastBall
      dimension={number('dimension', 25)}
      numberData={number('numberData', 1)}
      textData={text('textData', 'Last Ball')}
    />,
    () => (
      <LastBall
        dimension={number('dimension', 25)}
        numberData={number('numberData', 1)}
        textData={text('textData', 'Last Ball')}
      />
    )
  );
