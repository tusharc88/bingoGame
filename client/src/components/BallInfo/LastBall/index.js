import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import BallInfo from '../index';
import Tag from '../../Tag/index';

const LastBallInfoWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  margin-left: 25px;
`;

const LastBallTextWrapper = styled.div`
  font-size: ${props => `${props.dimension}px`};
  line-height: ${props => `${props.dimension * 2}px`};
  margin-right: 25px;
`;

const LastBall = ({ dimension, numberData, textData }) => (
  <BallInfo numberData={numberData} textData={textData}>
    {(number, textInfo) => (
      <LastBallInfoWrapper>
        <LastBallTextWrapper dimension={dimension}>
          {textInfo}
        </LastBallTextWrapper>
        <Tag hasBorder dimension={dimension}>
          {number}
        </Tag>
      </LastBallInfoWrapper>
    )}
  </BallInfo>
);

LastBall.propTypes = {
  dimension: PropTypes.number.isRequired,
  numberData: PropTypes.number.isRequired,
  textData: PropTypes.string.isRequired,
};

export default LastBall;
