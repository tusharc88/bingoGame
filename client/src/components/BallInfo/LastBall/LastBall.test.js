import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import LastBall from './index';

test('Renders with default props', () => {
  const tree = renderer
    .create(<LastBall dimension={15} numberData={1} textData="Last Ball" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
