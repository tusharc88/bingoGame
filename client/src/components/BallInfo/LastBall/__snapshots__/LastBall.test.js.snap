// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with default props 1`] = `
.c0 {
  width: 100%;
}

.c3 {
  height: 30px;
  width: 45px;
  background-color: #edf0f3;
  text-align: center;
  vertical-align: middle;
  font-size: 15px;
  line-height: 30px;
  border-radius: 5px;
  border: 2px solid #3ecedf;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-left: 25px;
}

.c2 {
  font-size: 15px;
  line-height: 30px;
  margin-right: 25px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Last Ball
    </div>
    <div
      className="c3"
    >
      1
    </div>
  </div>
</div>
`;
